### ISSUE TITLE
Allow one-sided one-touch-expandable split panes

### ISSUE THREAD

**Issue by Chrriis** [JFormDesigner/FlatLaf#355]
Here is a regular one-touch-expandable split pane:
![SplitPaneOneTouch_01](https://user-images.githubusercontent.com/1174202/125909799-90c325ff-f3e9-4305-8eae-c92ccb0eb6b4.png)

When we collapse, FlatLaf hides the arrow that is not applicable:
![SplitPaneOneTouch_02](https://user-images.githubusercontent.com/1174202/125909847-a862440f-af47-429e-aa30-5cea39b4ee9e.png)

This made me think of a general UI improvement. In many occasions, only one of the 2 sides makes sense to be collapsed. In the above case, there is absolutely no point in collapsing the right side.
It would be a nice improvement to allow one-sided one-touch-expandable split panes, probably through the use of a client property. We would get something like:
![SplitPaneOneTouch_03](https://user-images.githubusercontent.com/1174202/125910072-13904e36-9ce1-40be-be5e-fd26f231f9d0.png)

Test code, including a proposed client-property in a comment:
```Java
public static void main(String[] args) throws Exception {
	UIManager.setLookAndFeel(new FlatLightLaf());
	// UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
	JFrame frame = new JFrame();
	frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
	JPanel centerPane = new JPanel(new BorderLayout());
	centerPane.setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
	Object[][] data = new Object[][] { { "Cell 1_1", "Cell 1_2", Boolean.FALSE }, { "Cell 2_1", "Cell 2_2", Boolean.TRUE }, { "Cell 3_1", "Cell 3_2", Boolean.TRUE }, { "Cell 4_1", "Cell 4_2", Boolean.FALSE } };
	String[] titles = new String[] { "Column1", "Column2", "Column3" };
	JTable table = new JTable(data, titles);
	JSplitPane splitPane = new JSplitPane(JSplitPane.HORIZONTAL_SPLIT, true, new JScrollPane(new JTree()), new JScrollPane(table));
	splitPane.setOneTouchExpandable(true);
	//splitPane.putClientProperty("oneSided", "left");
	splitPane.setDividerLocation(100);
	centerPane.add(splitPane);
	frame.getContentPane().add(centerPane);
	frame.setSize(600, 400);
	frame.setVisible(true);
}
```

**Comment by DevCharly** [comment]
Very good idea 👍

**Comment by DevCharly** [comment]
This is now implemented in main branch for FlatLaf 2.2

![image](https://user-images.githubusercontent.com/5604048/165074456-fc2cb2ba-7536-428d-87ee-43bac30f9f3d.png)

To enable it use either:

~~~java
splitPane.putClientProperty( "JSplitPane.expandableSide", "left" );
~~~
or
~~~java
splitPane.putClientProperty( "JSplitPane.expandableSide", "right" );
~~~

Docs: https://www.formdev.com/flatlaf/client-properties/#JSplitPane
